src/adapters/permissions-manager.ts

```diff
@@ -0,0 +1,65 @@
+import { AllowListPermissionSetting, AppComponents, IPermissionsManager, Permission, PermissionType } from '../types'
+import { defaultPermissions } from '../logic/permissions-checker'
+import bcrypt from 'bcrypt'
+
+const saltRounds = 10
+
+export async function createPermissionsManagerComponent({
+  worldsManager
+}: Pick<AppComponents, 'worldsManager'>): Promise<IPermissionsManager> {
+  async function setPermissionType(
+    worldName: string,
+    permission: Permission,
+    type: PermissionType,
+    extras: any
+  ): Promise<void> {
+    const metadata = await worldsManager.getMetadataForWorld(worldName)
+
+    const extraOptions: any = {}
+    if (type === PermissionType.SharedSecret) {
+      extraOptions.secret = bcrypt.hashSync(extras.secret, saltRounds)
+    } else if (type === PermissionType.NFTOwnership) {
+      extraOptions.nft = extras.nft
+    } else if (type === PermissionType.AllowList) {
+      extraOptions.wallets = []
+    }
+
+    const permissions = metadata?.permissions || defaultPermissions()
+    permissions[permission] = { type, ...extraOptions }
+    await worldsManager.storePermissions(worldName, permissions)
+  }
+
+  async function addAddressToAllowList(worldName: string, permission: Permission, address: string): Promise<void> {
+    const metadata = await worldsManager.getMetadataForWorld(worldName)
+
+    const permissionSetting = metadata!.permissions[permission] as AllowListPermissionSetting
```

Comment:
why can't metadata be undefined?

Reply:
I also don't think you need the `as ...` if you move it after the check

Reply:
fixed